connectors/webhook/src/main/java/io/camunda/connector/inbound/signature/strategy/URLAndBodyEncodingStrategy.java

```diff
@@ -15,30 +15,45 @@
 import java.util.Map;
 
 public final class URLAndBodyEncodingStrategy implements HMACEncodingStrategy {
-  @Override
-  public byte[] getBytesToSign(final WebhookProcessingPayload payload) throws IOException {
-    return (payload.requestURL() + extractSignatureData(payload)).getBytes();
-  }
 
-  private static String extractSignatureData(final WebhookProcessingPayload payload)
-      throws IOException {
+  private static String extractSignatureData(final WebhookProcessingPayload payload) {
     if (payload.rawBody() == null || payload.rawBody().length == 0) {
       throw new NullPointerException(
           "Can't extract signature data from body, because body is null");
     }
     Map<String, String> signatureData =
-        HttpWebhookUtil.transformRawBodyToMap(
-            payload.rawBody(), HttpWebhookUtil.extractContentType(payload.headers()));
+        checkedCastToMap(
+            HttpWebhookUtil.transformRawBodyToObject(
+                payload.rawBody(), HttpWebhookUtil.extractContentType(payload.headers())),
+            String.class);
 
-    StringBuilder builder = new StringBuilder();
     List<String> sortedKeys = new ArrayList<>(signatureData.keySet());
     Collections.sort(sortedKeys);
 
+    StringBuilder builder = new StringBuilder();
+
     for (String key : sortedKeys) {
       builder.append(key);
       String value = signatureData.get(key);
       builder.append(value == null ? "" : value);
     }
     return builder.toString();
   }
+
+  private static <T> Map<T, T> checkedCastToMap(Object o, Class<T> tClass) {
```

Comment:
Looks like we don't have any other use cases than the `tClass` being a `String`. I'd suggest removing the generics then for visibility